### ISSUE TITLE
Causes lag when snapping windows and other random stuff when desktop is visible

### ISSUE THREAD

**Issue by GanerCodes** [y4my4my4m/kde-shader-wallpaper#30]
I have no idea why it happens but if I have shader desktop enabled snapping windows causes a big stutter to my system

**Comment by GanerCodes** [comment]
Note this seems to also occur on other wallpaper types that involve shaders

**Comment by GanerCodes** [comment]
Example:

https://user-images.githubusercontent.com/34655998/154803846-eaa1ffb7-fa19-4cea-a9c3-a5e1eadac922.mp4

**Comment by y4my4my4m** [comment]
@GanerCodes 
which version are you running? 2.0.1a?

also, is `Pause teh shader when covered by maximized or full-screen windows` on?
![image](https://user-images.githubusercontent.com/8145020/154865802-2bfde049-08be-4ef7-82fe-7b80cdc832c8.png)

If so, turn it off and see...


I've been working towards refactoring the whole thing, but my gut feeling is that the shader gets restarted many times for no good reason (because of the shader editing code). Maybe start `plasmashell` from a shell session so you can read the log output and see if it says `QML Shader restarted` everything it snaps to windows.

**Comment by GanerCodes** [comment]
It seems the shader restarts every time I let go of a window. There's also a kwin error that pops up if I go over the snaps too much. Same thing happens regardless of compatibility mode and full-screen pausing
[Example](https://ganer.xyz/s/544bb6fe865b45d8)  
  
Also thanks haha

**Comment by y4my4my4m** [comment]
@GanerCodes 

I'm not surer if this helps anything but i've applied a hotfix to fix this error:
![image](https://user-images.githubusercontent.com/8145020/158709273-96de1ec6-405d-40d6-91b3-0677ecafd937.png)

Try it from the master branch [how-to](https://github.com/y4my4my4m/kde-shader-wallpaper#install-from-source)

**Comment by GanerCodes** [comment]
Hey, I actually solved this issue by switching GPUs lol. Nvidia was causing an absolute plethora of issues on my system involving UI/UX and on AMD I have almost none.  

Also, on an unrelated note is there some way to make it so only fullscreen programs pause the shader? When coding I have a transparent window and I can see my Desktop however I'd rather the shader not run if I'm playing a fullscreen game.

**Comment by y4my4my4m** [comment]
@GanerCodes 
Yes, sorry the UI is crap right now. Working on it on the #25 PR but, if you make the window larger you should see this checkbox option:

![image](https://user-images.githubusercontent.com/8145020/158710439-39fe8f31-e7fe-4485-84fd-1e0b39f63f3a.png)

**Comment by GanerCodes** [comment]
I'm refering to pausing the shader only when covered with fullscreen windows, not maximised

**Comment by y4my4my4m** [comment]
@GanerCodes 

https://github.com/y4my4my4m/kde-shader-wallpaper/blob/db55dd3ab183d31d3c4cd6a5204b84dde176bb74/contents/ui/Components/ShaderEngine.qml#L272

Yes, this should be easy to do by removing this conditional: ` || appWindow.IsMaximized` and only keeping `appWindow.IsFullScreen`
Will add as a feature in the refactored version, for now you will have to edit it yourself and load the plugin from source.

**Comment by y4my4my4m** [comment]
Should be fixed in the V4 C++ version (although it is for plasma6...). Reopen if not.
